docs/query_execution_plan_in_TiSpark.md

```diff
@@ -0,0 +1,69 @@
+
+### `Range`&`RangeFilter`
+
+In the `WHERE`/`HAVING`/`ON` conditions, the TiSpark optimizer analyzes the result returned by the primary key query or the index key query. For example, these conditions might include comparison operators of the numeric and date type, such as `>`, `<`, `=`, `>=`, `<=`, and the character type such as `LIKE`.
+
+The `Range` in `CoprocessorRDD`(`FetchHandleRDD`) represents the range of scanning. `RangeFilter` indicates which expression(s) the range is made up of. If `RangeFilter` is empty, it indicates a full table scan or full index scan. `RangeFilter` generally appears when the query involves an index range, when query the expressions in the `RangeFilter` form the scanned range from left to right.
+
+### Selection
+
+The expression passed to COP/TiKV as selection expression without triggering a downgrade.
+
+## Understand EXPLAIN output in `RegionTaskExec`
+
+From the previous article, we know that `IndexLookUp` will perform two scanning operations, the first scan is `IndexRangeScan` and the second scan is `TableRowIDScan`. If the `TableRowIDScan` in `IndexLookUp` does too many queries on COP([TiKV Coprocessor](https://docs.pingcap.com/tidb/stable/tikv-overview#tikv-coprocessor))/TiKV, it can cause performance problems in COP/TiKV. To solve this problem a downgrading mechanism is introduced.
+
+The `IndexRangeScan` of `IndexLookUp` will return the data that meets the conditions, and then TiSpark will sort and aggregate the returned data to obtain the `regionTask` that needs to be done in the `TableRowIDScan`. If the number of `regionTask` is bigger than `downgradeThreshold`, a downgrade will be triggered. When a downgrade is triggered, the range of the second scan table will be changed from the returned value of the first scan index to all values between the minimum and maximum value of the first scan index, and the `filters` of the second scan will change to `downgradeFilters`(`downgradeFilters` is the same as if the execution plan is `TableScan`'s `filters`).
+
+> **`RegionTask`**
+>
+> For all returned data, all consecutive data in a region will be treated as a `regionTask`.
+>
+> For example like this the data returned in the first stage are 1, 3, 4, 5 and 1, 3, 4 are in the same region and 5 is in another region. Since 1 and 3, 4 are not contiguous, 1 is a `regionTask`, and since 3, 4 and 5 are not in a region, 3, 4 is a `regionTask` and 5 is a `regionTask`. The number of `regionTask`  is three.
+
+### `downgradeThreshold`
+
+The threshold value that triggers a downgrade. The downgrade is triggered when the number of `RegionTask` exceeds the `downgradeThreshold`.
+
+### `downgradeFilter`
+
+The expression passed to COP/TiKV as selection expression when triggering a downgrade.
```

Comment:
we need to tell the user that the downgradeFilter is the filter that may downgrade, it may not downgrade. right?